src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.ts

```diff
@@ -0,0 +1,245 @@
+  @Input() pods: any;
+  @Input() mini: boolean;
+  @Input() desiredReplicas: number;
+  @Input() idled: boolean;
+
+  chartId = uniqueId('deployments-donut-chart');
+  podStatusData: any;
+  total: number;
+
+  private phases = ['Running',
+    'Not Ready',
+    'Warning',
+    'Error',
+    'Pulling',
+    'Pending',
+    'Succeeded',
+    'Terminating',
+    'Unknown'];
+
+  private config: any;
+  private chart: any;
+
+
+  private debounceUpdate: (countByPhase: any) => void;
+  private prevPodPhaseCount: any;
+
+  constructor() { }
+
+  ngOnInit(): void {
+    this.config = {
+      type: 'donut',
+      bindto: '#' + this.chartId,
+      donut: {
+        expand: false,
+        label: {
+          show: false
+        },
+        width: this.mini ? 5 : 10
+      },
+      size: {
+        height: this.mini ? 45 : 150,
+        width: this.mini ? 45 : 150
+      },
+      legend: {
+        show: false
+      },
+      tooltip: {
+        format: {
+          value: function (value, ratio, id) {
+            if (!value) {
+              return undefined;
+            }
+            if (id === 'Empty') {
+              return undefined;
+            }
+
+            return value;
+          }
+        }
+      },
+      transition: {
+        duration: 350
+      },
+      data: {
+        type: 'donut',
+        groups: [this.phases],
+        order: null,
+        colors: {
+          Empty: '#ffffff',
+          Running: '#00b9e4',
+          'Not Ready': '#beedf9',
+          Warning: '#f39d3c',
+          Error: '#d9534f',
+          Pulling: '#d1d1d1',
+          Pending: '#ededed',
+          Succeeded: '#3f9c35',
+          Terminating: '#00659c',
+          Unknown: '#f9d67a'
+        },
+        selection: {
+          enabled: false
+        }
+      }
+    };
+
+    if (this.mini) {
+      this.config.padding = {
+        top: 0,
+        right: 0,
+        bottom: 0,
+        left: 0
+      };
+    }
+
+    this.debounceUpdate = debounce(this.updateChart, 350, { maxWait: 500 });
+  }
+
+  ngDoCheck(): void {
+    let podPhaseCount = this.countPodPhases();
+    if (!isEqual(this.prevPodPhaseCount, podPhaseCount)) {
+      this.prevPodPhaseCount = podPhaseCount;
+      this.debounceUpdate(podPhaseCount);
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (!changes.desiredReplicas.firstChange || !changes.idled.firstChange) {
+      this.updatePodCount();
+    }
+  }
+
+  ngOnDestroy(): void {
+    if (this.chart) {
+      this.chart = this.chart.destroy();
+    }
+  }
+
+  private updatePodCount(): void {
+    let pods = reject(this.pods, { status: { phase: 'Failed' } });
+    let total = size(this.pods);
+
+    if (this.mini) {
+      this.total = total;
+      return;
+    }
+
+    let smallText: string;
+    if (isNaN(this.desiredReplicas) || this.desiredReplicas === total) {
+      smallText = (total === 1) ? 'pod' : 'pods';
+    } else {
+      smallText = 'scaling to ' + this.desiredReplicas + '...';
+    }
```

Comment:
small nit: `scaling to ${this.desiredReplicas}...` might be cleaner

Reply:
```
smallText = 'scaling to ${this.desiredReplicas}...';
```

didn't work for me. If the code isn't wrong and the access of the value `this.desiredReplicas` is later on, I don't think it will work since it doesn't have access to the same scope.

Reply:
Did you use those single quotes or backtick quotes? You need backticks for string interpolation. It's just syntax sugar over the same thing you're doing with the + concat. I don't think it has any effect on temporality of when the value is read.